File: lib/src/main/java/one/hyro/lib/utils/ApiRequest.java
package one.hyro.lib.utils;

import com.google.gson.JsonObject;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class ApiRequest {
    public static void POST(String url, JsonObject body) {
        try {
            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(new URI(url))
                    .header("Content-Type", "application/json")
                    .POST(HttpRequest.BodyPublishers.ofString(body.toString()))
                    .build();

            client.send(request, HttpResponse.BodyHandlers.ofString());
        } catch (Exception e) {}
    }

    public static String GET(String url, JsonObject body) {
        try {
            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(new URI(url))
                    .header("Content-Type", "application/json")
                    .GET()
                    .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            return response.body();
        } catch (Exception e) {
            return null;
        }
    }

    public static void DELETE(String url) {
        try {
            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(new URI(url))
                    .header("Content-Type", "application/json")
                    .DELETE()
                    .build();

            client.send(request, HttpResponse.BodyHandlers.ofString());
        } catch (Exception e) {}
    }

    public static void PATCH(String url, JsonObject body) {
        try {
            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(new URI(url))
                    .header("Content-Type", "application/json")
                    .method("PATCH", HttpRequest.BodyPublishers.ofString(body.toString()))
                    .build();

            client.send(request, HttpResponse.BodyHandlers.ofString());
        } catch (Exception e) {}
    }
}

File: lib/src/main/kotlin/api/ApiRequest.kt
package api

import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody

object ApiRequest {
    private val client = OkHttpClient()
    private val JSON = "application/json".toMediaType()

    fun get(url: String): String {
        val request = Request.Builder()
            .url(url)
            .build()

        client.newCall(request).execute().use { response ->
            return response.body!!.string()
        }
    }

    fun post(url: String, body: String) {
        val request = Request.Builder()
            .url(url)
            .post(body.toRequestBody(JSON))
            .build()

        client.newCall(request).execute().use { response ->
            println(response.body!!.string())
        }
    }

    fun put(url: String, body: String) {
        val request = Request.Builder()
            .url(url)
            .put(body.toRequestBody(JSON))
            .build()

        client.newCall(request).execute().use { response ->
            println(response.body!!.string())
        }
    }

    fun delete(url: String) {
        val request = Request.Builder()
            .url(url)
            .delete()
            .build()

        client.newCall(request).execute().use { response ->
            println(response.body!!.string())
        }
    }

    fun patch(url: String, body: String) {
        val request = Request.Builder()
            .url(url)
            .patch(body.toRequestBody(JSON))
            .build()

        client.newCall(request).execute().use { response ->
            println(response.body!!.string())
        }
    }
}